### DISCUSSION TITLE
Problem with is not pattern matching

### DISCUSSION THREAD

**Discussion by msedi** [dotnet/csharplang#4433] [Q&A] [answered]
I have the following lines of code, where exp is a regular expression:

```cs
  ConvolutionKernel.Filter = (o) =>
            {
                if (exp == null)
                    return true;

                if (!(o is KernelEntry aKernelEntry))
                    return false;

                return (exp.IsMatch(aKernelEntry.Name));
            };
```

Everything works fine, until I switch to the new` is not `pattern matching syntax

```cs
   ConvolutionKernel.Filter = (o) =>
            {
                if (exp == null)
                    return true;

                if (o is not KernelEntry aKernelEntry)
                    return false;

                return (exp.IsMatch(aKernelEntry.Name));
            };
```

If condition `if (o is not KernelEntry aKernelEntry)` is passed, because o is a KernelEntry an exception is thrown that aKernelEntry is null. Are there any explanations for this behavior. I'm using this pattern all over my code base but only at the special location I have this problem?

**Comment by bernd5** [comment]
Could you create a minimal sample to reproduce your issue, please?

**Reply by msedi** [reply]
@bernd5: Thanks for responding. I would if I could. The problem is that it's hard to isolate. Since on every other place that uses the same methods works. I will try to see what I can do to recreate the problem on a minimal example.

**Comment by huoyaoyuan** [comment, marked as answer]
This sounds like the codegen bug for `is not` in 16.8?

In this case, you may try VS16.9 Preview or sharplab master to confirm.

**Reply by 333fred** [reply]
Yes, https://github.com/dotnet/roslyn/issues/49262

**Reply by msedi** [reply]
@huoyaoyuan + @333fred: Thanks for this tip. I will try with VS16.9. Unfortunately VS16.9 is extremely unstable at the moment. So for now I will just leave the code as it is. If VS16.9 is out, I will try again. Thanks for your time.
